Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+
+        /// Starts the location data source and awaits location and satellite updates.
+        /// - Parameter usingMockedData: Indicates that the location datasource should use mocked data.
+        func start(usingMockedData: Bool = false) {
+            // Set NMEA location data source for location display.
+            locationDisplay.dataSource = nmeaLocationDataSource
+            // Set buttons states.
+            isSourceButtonDisabled = true
+            isResetButtonDisabled = false
+            
+            // Start the data source and location display.
+            if usingMockedData {
+                mockNMEADataSource.start(with: nmeaLocationDataSource)
+            }
+            
+            // Set the autopan mode to `.recenter`
+            autoPanMode = .recenter
+
+            Task {
+                // Start the data source
+                try await locationDisplay.dataSource.start()
+            }
+            
+            // Kick off tasks to monitor autoPan, locations, and satellites.
+            observationTasks.append(
+                contentsOf: [
+                    observeAutoPanTask,
+                    observeLocationsTask,
+                    observeSatellitesTask
+                ]
+            )
+        }
+        
+        /// A detached task observing location display autoPan changes.
+        var observeAutoPanTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("mode in locationDisplay.$autoPanMode task ending") }
+                for await mode in locationDisplay.$autoPanMode {
+                    await MainActor.run {
+                        isRecenterButtonDisabled = nmeaLocationDataSource == nil || mode == .recenter
+                    }
+                }
+            }
+        }
+        
+        /// A detached task observing location data source location changes.
+        var observeLocationsTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("location in nmeaLocationDataSource.locations task ending") }
+                for await location in nmeaLocationDataSource.locations {
+                    guard let nmeaLocation = location as? NMEALocation,
+                          nmeaLocationDataSource.status == .started else { return }
+                    let horizontalAccuracy = Measurement(
+                        value: nmeaLocation.horizontalAccuracy,
+                        unit: UnitLength.meters
+                    )
+                    let verticalAccuracy = Measurement(
+                        value: nmeaLocation.verticalAccuracy,
+                        unit: UnitLength.meters
+                    )
+                    let accuracyText = String(
+                        format: "Accuracy - Horizontal: %@; Vertical: %@",
+                        distanceFormatter.string(from: horizontalAccuracy),
+                        distanceFormatter.string(from: verticalAccuracy)
```

Comment:
I recommend using [FormatStyle](https://developer.apple.com/documentation/foundation/formatstyle) instead of a formatter. I recall hearing someone saying that formatters are not preferred starting from iOS 16, so I logged a [card](https://github.com/Esri/arcgis-maps-sdk-swift-samples/projects/1#card-88788528) to take a look in the future.

```swift
let formatStyle = Measurement<UnitLength>.FormatStyle.measurement(
    width: .abbreviated,
    usage: .general,
    numberFormatStyle: .number.precision(.fractionLength(1))
)
let _ = verticalAccuracy.formatted(formatStyle)
```

The old [`naturalScale`](https://developer.apple.com/documentation/foundation/measurementformatter/unitoptions) API doesn't have a equivalent in the [new ones](https://developer.apple.com/documentation/foundation/measurementformatunitusage), so I assume providedUnit = [asProvided](https://developer.apple.com/documentation/foundation/measurementformatunitusage/3767012-asprovided), naturalScale = [general](https://developer.apple.com/documentation/foundation/measurementformatunitusage/3767015-general), temperatureWithoutUnit = [weather](https://developer.apple.com/documentation/foundation/measurementformatunitusage/3767024-weather).

More examples [here](https://goshdarnformatstyle.com/measurement-style/).